R/sw_Vegetation.R

```diff
@@ -928,286 +620,661 @@ get_season_description_v2 <- function(x) {
+
+  # Season sequences
+  #   - must be continuous for "continuous adjustment sections"
+  #   - make sure season ellipses' vertices are included in sequence
+  #   - increase density of points near season ellipses' vertices
+  #     (where uniroot tends to fail and vegetation curve often changes fast)
+  tmp <- pi / 2
+  seq_angle1 <- sort(unique(c(
+    c(-1, 1) * tmp,
+    seq_len(Nr) / Nr * 2 * pi
+  )))
+  seq_angle2 <- seq_len(2 * Nr) / Nr * pi
+
+  seq_cold1 <- fun_cold(seq_angle1)
+  ids_cold1 <- which(
+    seq_cold1[, 2] <= ctr_cold[2] &
+      seq_cold1[, 1] >= xrange_cold[1] &
+      seq_cold1[, 1] <= xrange_cold[2]
+  )
+  seq_cold2 <- fun_cold(seq_angle2)
+  ids_cold2 <- which(
+    seq_cold2[, 2] <= ctr_cold[2] &
+      seq_cold2[, 1] >= xrange_cold2[1] &
+      seq_cold2[, 1] <= xrange_cold2[2]
+  )
+  seq_cold <- unique(rbind(seq_cold1[ids_cold1, ], seq_cold2[ids_cold2, ]))
+  seq_cold <- seq_cold[order(seq_cold[, 1]), ]
+
+  seq_warm1 <- fun_warm(seq_angle1)
+  ids_warm1 <- which(
+    seq_warm1[, 2] >= ctr_warm[2] &
+      seq_warm1[, 1] >= xrange_warm[1] &
+      seq_warm1[, 1] <= xrange_warm[2]
+  )
+  seq_warm2 <- fun_warm(seq_angle2)
+  ids_warm2 <- which(
+    seq_warm2[, 2] >= ctr_warm[2] &
+      seq_warm2[, 1] >= xrange_warm2[1] &
+      seq_warm2[, 1] <= xrange_warm2[2]
+  )
+  seq_warm <- unique(rbind(seq_warm1[ids_warm1, ], seq_warm2[ids_warm2, ]))
+  seq_warm <- seq_warm[order(seq_warm[, 1]), ]
+
+  n_cold <- nrow(seq_cold)
+  n_warm <- nrow(seq_warm)
+  Nadj <- n_cold + n_warm
+
+
+  # Information about the points that define the lines which are used
+  # to intersect values for adjustments
+  # columns:
+  #   1:2 = x/y-coords of center of cold-/warm-season ellipses
+  #   3:4 = x/y-coords of points along cold-/warm-season ellipses
+  #   5:6 = min/max of x-coord under influence of cold-/warm-season ellipses
+  #   7 = x-extent of influence around center of cold-/warm-season ellipses
+  ptadj <- matrix(NA, nrow = Nadj, ncol = 7)
+
+  tmp <- seq_len(n_cold)
+  ptadj[tmp, 1:2] <- rep(ctr_cold, each = n_cold)
+  ptadj[tmp, 3:4] <- seq_cold
+  ptadj[tmp, 5:6] <- rep(xrange_cold, each = n_cold)
+  ptadj[tmp, 7] <- infl_cold
+
+  tmp <- n_cold + seq_len(n_warm)
+  ptadj[tmp, 1:2] <- rep(ctr_warm, each = n_warm)
+  ptadj[tmp, 3:4] <- seq_warm
+  ptadj[tmp, 5:6] <- rep(xrange_warm, each = n_warm)
+  ptadj[tmp, 7] <- infl_warm
+
+
+  #------ Step 2) Normalize (shift and scale) all values
+  # so that they fit into the season ellipses
+
+  # Normalize time of year values (on x-axis) into [0, 1]
+  mon_norm <- mon / nmon
+
+  # Normalize temperature values (on y-axis) into [0, 0.5]
+  temp_zero <- -min(ref_temp, target_temp)
+  tmp1 <- temp_zero + ref_temp
+  tmp2 <- temp_zero + target_temp
+  temp_2max <- max(tmp1, tmp2) / vals_std[2]
+
+  ref_temp_norm <- tmp1 / temp_2max
+  ref_temp_norm_mean <- mean(ref_temp_norm)  # nolint
+  target_temp_norm <- tmp2 / temp_2max  # nolint
```

Comment:
Unclear what is meant here by `# nolint`

Reply:
See https://github.com/jimhester/lintr#project-configuration
`# nolint` is the default "regex pattern for lines to exclude from linting" by the `lintr` R package